hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java

```diff
@@ -698,11 +699,12 @@ public void loadAll(Set<K> keys, boolean replaceExistingValues) {
         MapKeySetRequest request = new MapKeySetRequest(name);
         MapKeySet mapKeySet = invoke(request);
         Set<Data> keySetData = mapKeySet.getKeySet();
-        Set<K> keySet = new HashSet<K>(keySetData.size());
+        InflatableSet<K> keySet = new InflatableSet<K>(keySetData.size());
         for (Data data : keySetData) {
             final K key = toObject(data);
```

Comment:
Sounds better but with that approach, isn't there a possibility that we are creating too many lazy-entries even we don't need to? Imagining a scenario that only needs to check the size of the returned key-set, in that case there is no need a set pre-populated with lazy-entries, i am expecting an approach which creates lazy-entries on demand, maybe i misunderstood your approach...